client/src/deepscanToken.ts

```diff
@@ -0,0 +1,122 @@
+
+export class DeepscanToken {
+  private readonly _secretStorage: vscode.SecretStorage;
+  private tokenName: string;
+  private serverUrl: string;
+  private tokenRenerateUrl: string;
+
+  constructor(context: vscode.ExtensionContext, serverUrl: string) {
+    this._secretStorage = context.secrets;
+    this._registerCommands();
+    context.subscriptions.push(context.secrets.onDidChange((e) => this._handleSecretChange(e)));
+    this.tokenName = 'deepscan-token';
+    this.serverUrl = serverUrl;
+    this.tokenRenerateUrl = `${serverUrl}/auth/git?redirect_uri=/dashboard/#view=account-settings`;
+  }
+
+  private _validateToken(token: string) {
+   if (!token) {
+      vscode.window.showWarningMessage(`Sorry, DeepScan access token for VS Code cannot be blank.`);
+      return false;
+   }
+    return true;
+  }
+
+  setToken(token: string) {
+    this._secretStorage.store(this.tokenName, token);
+  }
+
+  async getToken() {
+    return await this._secretStorage.get(this.tokenName);
+  }
+
+  deleteToken() {
+    this._secretStorage.delete(this.tokenName);
+  }
+
+  private async _handleSecretChange(e: vscode.SecretStorageChangeEvent) {
+    vscode.commands.executeCommand('deepscan.updateToken');
+  }
+
+  private _registerCommands() {
+    vscode.commands.registerCommand('deepscan.setToken', async () => {
+      let tokenInput: string = await vscode.window.showInputBox({
+          title: "Configure DeepScan Access Token For VS Code"
+      }) ?? '';
+      tokenInput = tokenInput.trim();
+      if (this._validateToken(tokenInput)) {
+        this.setToken(tokenInput);
+        vscode.window.showInformationMessage(`DeepScan access token has successfully configured.`);
+      }
+    });
+    vscode.commands.registerCommand('deepscan.deleteToken', async () => {
+      const token = await this.getToken();
+      if (token) {
+        const deleteAction = 'Delete';
+        const cancel = 'Cancel';
+        const message = `Are you sure you want to delete your DeepScan access token? The DeepScan extension will no longer be able to inspect.`;
+        const selected = await vscode.window.showWarningMessage(message, deleteAction, cancel);
+        if (selected === deleteAction) {
+          this.deleteToken();
+          vscode.window.showInformationMessage(`Deepscan access token is successfully deleted.`);
+        }
+      } else {
+        vscode.window.showInformationMessage(`Sorry, DeepScan access token doesn't exist.`);
+      }
+    });
+  }
+
+  async showOneOffTokenNotification() {
+    const token = await this.getToken();
+    if (token) {
+        return;
+    }
+
+    const generate = 'Generate';
+    const neverShowAgain = 'Don\'t show again';
+    const message = `DeepScan access token is required for using the DeepScan extension.
+    The token was introduced as a step towards providing personalization features.
+    You can generate a new token by clicking on the button below to go to DeepScan Account Settings page.`;
```

Comment:
```suggestion
    const message =
      'An access token is required for using the DeepScan extension. ' +
      'DeepScan server uses the token to provide reliable and managed inspection of your code.';
```

Multiline template literal 사용하면 newline 문자가 포함되게 되는데 차후에 VS Code에서 이를 보여주는 방식이 바뀔 수도 있을 듯 합니다.
따라서 그냥 space로 문장 구분하는 것이 좋을 듯 합니다.